### PR TITLE
Added search parameter feature for products

### PR DESCRIPTION
**Pull request by kubraa16**

### CODE REVIEW

File: src/store/reducers/productSlice.js [hunk 1]
```diff
@@ -37,23 +52,48 @@ const productsSlice = createSlice({
...
 export const fetchProductsData = createAsyncThunk(
   "products/fetchProductsData",
-  async (params, { getState, dispatch }) => {
-    const { page, limit } = getState().products;
+  async (params, { dispatch }) => {
+    const { page, limit, skip } = params;
+
     try {
       dispatch(setLoading(true));
+      let response;
```
Comment (PoojaVishnoi18) [inline src/store/reducers/productSlice.js]: use const

File: src/components/products/CustomTab.jsx [hunk 2]
```diff
@@ -36,18 +36,28 @@ const buttonConfig = {
...
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
   useEffect(() => {
     dispatch(fetchCategoryData());
-  }, []);
+  }, [dispatch]);
+
+  const handleCategorySelect = (categorySlug) => {
+    dispatch(setActiveCategory(categorySlug));
```
Comment (PoojaVishnoi18) [inline src/components/products/CustomTab.jsx]: remove this 
activeCategory is not getting used anywhere 
remove the set dispatch related to activeCategory from everywhere

File: src/store/reducers/productSlice.js (hunk 1 above)
Comment (kubraa16) [reply src/store/reducers/productSlice.js]: Done.

File: src/components/products/CustomTab.jsx (hunk 2 above)
Comment (kubraa16) [reply src/components/products/CustomTab.jsx]: Done.

**Pull request merged**